src/token/Token.sol

```diff
@@ -31,17 +31,17 @@ contract Token is IToken, VersionedContract, UUPS, Ownable, ReentrancyGuard, ERC
     ///                                                          ///
 
     /// @notice Reverts if caller is not an authorized minter
-    modifier onlyMinter() {
-        if (!minter[msg.sender]) {
-            revert ONLY_MINTER();
+    modifier onlyAuctionOrMinter() {
+        if (msg.sender != settings.auction && !minter[msg.sender]) {
+            revert ONLY_AUCTION_OR_MINTER();
         }
 
         _;
     }
 
-    modifier onlyMinterWithToken(uint256 tokenId) {
-        if (!minter[msg.sender]) {
-            revert ONLY_MINTER();
+    modifier onlyAuctionOrMinterWithToken(uint256 tokenId) {
```

Comment:
Could you just combine the two modifiers on burn? Or perhaps use the above modifier for burn then check the ownership within the burn function and not a modifier since it's only being used in one place.